projects/map-maplibre/src/lib/maplibre-layers.helpers.ts

```diff
@@ -249,28 +261,32 @@ export function createGeojsonLayer(l: ukisVectorLayer) {
 
 export function createLayersFromGeojsonTypes(feature: GeoJSONFeature, l: ukisLayer, index?: number) {
     let layer: LayerSpecification = {} as never;
-    const style = {
-        fill: {
-            color: feature?.properties?.fill || 'rgba(255,255,255,0.4)',
-        },
-        stroke: {
-            color: feature?.properties?.stroke || '#3399CC',
-            width: 1.25,
-        },
-        circle: {
-            radius: 5
-        }
-    };
 
-    switch (feature.geometry.type) {
+    // https://wiki.openstreetmap.org/wiki/Geojson_CSS
+    // https://github.com/mapbox/simplestyle-spec/tree/master/1.1.0
+    const defaultStyle = {
+        fill: 'rgba(255,255,255,0.4)',
+        stroke: '#3399CC',
+        'stroke-width': 1.25,
+        'circle-radius': 5
+    } as const;
+
+    const getFeaturePropStyle = <T>(prop: keyof typeof defaultStyle) => {
+        const style: DataDrivenPropertyValueSpecification<T> = ['coalesce', ['get', prop], defaultStyle[prop]];
```

Comment:
@indus maybe you can check if this is the correct expression to get the property from the features, and if there is no such property, use a default from the `defaultStyle`?

Reply:
Just by looking at it it seems ok. I have no chance to test this. Did you faced any problems with it?

Reply:
maybe a function like 

``` TS

const coalesce= <T>(key: keyof typeof defaultStyle, default?: string) => {
        const style: DataDrivenPropertyValueSpecification<T> = ['coalesce', ['get', key], default || defaultStyle[key]];
//...
}

coalesce('fill')
coalesce('fill', '#f00')
```

would be more versatile and descriptive?!